cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/indexer/SchemaIndexerTest.java

```diff
@@ -0,0 +1,101 @@
+
+/**
+ * Test {@link SchemaIndexer}.
+ */
+public class SchemaIndexerTest {
+
+  @Test
+  public void testSimpleSchema() throws Exception {
+    Schema simpleSchema = Schema.of(Schema.Type.INT);
+    Set<String> expected = Collections.emptySet();
+    SchemaIndexer indexer = new SchemaIndexer();
+    Id.DatasetInstance datasetInstance = Id.DatasetInstance.from("ns1", "ds1");
+    Set<String> actual = indexer.getIndexes(new MetadataEntry(datasetInstance, "schema", simpleSchema.toString()));
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testSimpleRecord() throws Exception {
+    Schema simpleSchema = Schema.recordOf("record1",
+                                          // String x
+                                          Schema.Field.of("x", Schema.of(Schema.Type.STRING)),
+                                          // String[] y
+                                          Schema.Field.of("y", Schema.arrayOf(Schema.of(Schema.Type.STRING))),
+                                          // Map<byte[],double> z
+                                          Schema.Field.of("z", Schema.mapOf(Schema.of(Schema.Type.BYTES),
+                                                                            Schema.of(Schema.Type.DOUBLE))));
+    Set<String> expected = ImmutableSet.of("record1", "record1:RECORD", "x", "x:STRING", "y", "y:ARRAY", "z", "z:MAP");
+    SchemaIndexer indexer = new SchemaIndexer();
+    Id.DatasetInstance datasetInstance = Id.DatasetInstance.from("ns1", "ds1");
+    Set<String> actual = indexer.getIndexes(new MetadataEntry(datasetInstance, "schema", simpleSchema.toString()));
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testComplexRecord() throws Exception {
+    Schema complexSchema = Schema.recordOf(
+      "record1",
+      Schema.Field.of(
+        "map1",
+        Schema.mapOf(
+          Schema.recordOf("record21",
+                          // String x
+                          Schema.Field.of("x", Schema.of(Schema.Type.STRING)),
+                          // String[] y
+                          Schema.Field.of("y", Schema.arrayOf(Schema.of(Schema.Type.STRING))),
+                          // Map<byte[],double> z
+                          Schema.Field.of("z", Schema.mapOf(Schema.of(Schema.Type.BYTES),
+                                                            Schema.of(Schema.Type.DOUBLE)))),
+          Schema.arrayOf(Schema.recordOf(
+            "record22",
+            Schema.Field.of("a",
+                            // Map<array<byte[]>, Map<boolean,byte[]> a
+                            Schema.mapOf(Schema.arrayOf(Schema.of(Schema.Type.BYTES)),
+                                         Schema.mapOf(Schema.of(Schema.Type.BOOLEAN),
+                                                      Schema.of(Schema.Type.BYTES)))
+            )))
+        )),
+      Schema.Field.of("i", Schema.unionOf(Schema.of(Schema.Type.INT), Schema.of(Schema.Type.NULL))),
```

Comment:
nit: This can be ``Schema.nullableOf(Schema.Type.INT)``, right? or are you testing something else here?